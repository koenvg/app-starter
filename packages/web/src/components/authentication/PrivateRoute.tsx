import { useAuthentication } from "./AuthenticationProvider";
import React, { SFC } from "react";
import { Route, RouteProps, Redirect } from "react-router-dom";

export type Props = RouteProps;
export const PrivateRoute: SFC<Props> = ({ children, ...rest }) => {
  const { isAuthenticated } = useAuthentication();

  return (
    <Route {...rest}>
      {({ location }) => {
        if (!isAuthenticated) {
          return (
            <Redirect
              to={{
                pathname: "/login",
                state: { from: location },
              }}
            />
          );
        }
        return children;
      }}
    </Route>
  );
};

PrivateRoute.displayName = "PrivateRoute";
