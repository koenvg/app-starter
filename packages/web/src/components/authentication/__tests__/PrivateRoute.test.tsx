const useAuthenticationMock = jest.fn().mockReturnValue({ isAuthenticated: false });
jest.mock("../AuthenticationProvider", () => ({
  useAuthentication: useAuthenticationMock,
}));
import { PrivateRoute } from "../PrivateRoute";
import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter, Route, Switch } from "react-router-dom";

describe("PrivateRoute", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should redirect to the login page when the user is not authenticated", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Switch>
          <Route path="/login">
            <div>Login Route</div>
          </Route>
          <PrivateRoute path="/" exact>
            <div>Private Route</div>
          </PrivateRoute>
        </Switch>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render the component for the route when the user is authenticated", () => {
    useAuthenticationMock.mockReturnValueOnce({ isAuthenticated: true });
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <Switch>
          <PrivateRoute path="/" exact>
            <div>Private Route</div>
          </PrivateRoute>
          <Route path={"/login"} render={() => <div>{"/login"}</div>} />
        </Switch>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
