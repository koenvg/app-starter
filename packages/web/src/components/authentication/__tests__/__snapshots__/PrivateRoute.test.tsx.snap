// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute should redirect to the login page when the user is not authenticated 1`] = `
<div>
  <div>
    Login Route
  </div>
</div>
`;

exports[`PrivateRoute should render the component for the route when the user is authenticated 1`] = `
<div>
  <div>
    Private Route
  </div>
</div>
`;
